### DISCUSSION TITLE
Not been able to save raster into file or display through a MemoryStream and BitmapImage.

### DISCUSSION THREAD

**Discussion by miker1423** [MaxRev-Dev/gdal.netcore#117] [Q&A] [answered]
Running the [example at GDAL repository](https://github.com/OSGeo/gdal/blob/master/swig/csharp/apps/GDALDatasetRasterIO.cs) throws "A generic error occurred in GDI+." when trying to save the Bitmap to the file.  I've tried to save to a memory stream and displaying the image in a WPF application with the following code, but it shows the same behaivour throwing the exception when saving to the `MemoryStream`: 
```CSharp
using var stream = new MemoryStream();
bitmap.Save(stream, ImageFormat.Bmp);
stream.Position = 0;
DisplayImage = new BitmapImage();
DisplayImage.BeginInit();
DisplayImage.StreamSource = stream;
DisplayImage.CacheOption = BitmapCacheOption.OnLoad;
DisplayImage.EndInit();
```

Has anyone had this same issue? How can I debug further given that GDI+ is not great with errors?

Thanks!

**Comment by MaxRev-Dev** [comment, marked as answer]
I have no clue as it's not related to GDAL, but this could help.
https://stackoverflow.com/questions/15862810/a-generic-error-occurred-in-gdi-in-bitmap-save-method

Especially, this answer (non-existent folder in path or missing permissions)
https://stackoverflow.com/a/23695771

**Comment by miker1423** [comment]
I'm aware it's not a problem directly related with GDAL. Still wondering if someone could know why this might be happening and could help. I'm saving directly to a `MemoryStream`, therefore no folder or file is involved in the process.
